packages/pluggableWidgets/heatmap-chart-web/src/__tests__/HeatMap.spec.tsx

```diff
@@ -0,0 +1,237 @@
+import { createElement } from "react";
+import { ChartWidget } from "@mendix/shared-charts";
+import {
+    buildListExpression,
+    dynamicValue,
+    EditableValueBuilder,
+    ListAttributeValueBuilder,
+    ListValueBuilder
+} from "@mendix/piw-utils-internal";
+import { mount, ReactWrapper } from "enzyme";
+import { HeatMap } from "../HeatMap";
+import Big from "big.js";
+import { ObjectItem } from "mendix";
+import { HeatMapContainerProps } from "../../typings/HeatMapProps";
+
+jest.mock("@mendix/shared-charts", () => ({
+    ChartWidget: jest.fn(() => null)
+}));
+
+describe("The HeatMap widget", () => {
+    function renderHeatMap(props: Partial<HeatMapContainerProps>): ReactWrapper {
+        return mount(
+            <HeatMap
+                name="line-chart-test"
+                class="line-chart-class"
+                showLegend={false}
+                developerMode="basic"
+                widthUnit="percentage"
+                width={0}
+                heightUnit="pixels"
+                height={0}
+                customLayout=""
+                customConfigurations=""
+                customSeriesOptions=""
+                seriesDataSource={ListValueBuilder().simple()}
+                seriesName={buildListExpression("name")}
+                seriesValueAttribute={new ListAttributeValueBuilder<Big>().build()}
+                enableThemeConfig={false}
+                scaleColors={[]}
+                showScale
+                horizontalSortOrder="asc"
+                verticalSortOrder="asc"
+                smoothColor={false}
+                showValues={false}
+                valuesColor=""
+                {...setupBasicAttributes()}
+                {...props}
+            />
+        );
+    }
+
+    it("visualizes data as a heatmap chart", () => {
```

Comment:
Can we add a test case when there is no data?

Reply:
sure